### ISSUE TITLE
ERD diagram of mermaid is not rendered

### ISSUE THREAD

**Issue by tim-hub** [laurent22/joplin#3108]
Play round the mermaid diagram, found the ERD is not rendered.

<img width="779" alt="image" src="https://user-images.githubusercontent.com/2884320/79838876-33315380-83a3-11ea-9f5a-827e44a03138.png">

## Environment

Joplin version: 1.0.201
Platform: Mac OS
OS specifics: (prod, darwin) Mac OS Catalina

## Steps to reproduce

1. follow the screen shot, create a ERD diagram

## Describe what you expected to happen

the ERD should be rendered in preview windows.

## Logfile

please see the screenshot above.

**Comment by laurent22** [comment]
Why not provide the Markdown you used so that we can replicate the issue?

**Comment by coderrsid** [comment]
## Markdown to recreate the issue
```
# erd
```mermaid
erDiagram
	CUSTOMER ||--o{ ORDER : places
	ORDER ||--|{  LINE-ITEM : contains
	CUSTOMER }|..|{ DELIVERY-ADDRESS : uses
```

**Comment by coderrsid** [comment]
Although @tim-hub , these are experimental features of [Mermaid-js-erd](https://github.com/mermaid-js/mermaid/tree/develop/docs#entity-relationship-diagram---exclamation-experimental)

![image](https://user-images.githubusercontent.com/35633575/79857975-4288bf00-83ec-11ea-9bf5-52332fe08028.png)

**Comment by tessus** [comment]
It's possible that our Mermaid version does not even have that feature yet.

**Comment by coderrsid** [comment]
> It's possible that our Mermaid version does not even have that feature yet.

You're perfectly right, tessus. This feature is added to v8.5.0 and we're using v8.4.6 . Everyone can see below in the changelog of mermaid-js **Features** section.

![image](https://user-images.githubusercontent.com/35633575/79864045-64873f00-83f6-11ea-9c45-37b5673c107e.png)

**Comment by tim-hub** [comment]
> Why not provide the Markdown you used so that we can replicate the issue?

just forgot

**Comment by tim-hub** [comment]
> ## Markdown to recreate the issue
> 
> ```
> # erd
> ```mermaid
> erDiagram
> 	CUSTOMER ||--o{ ORDER : places
> 	ORDER ||--|{  LINE-ITEM : contains
> 	CUSTOMER }|..|{ DELIVERY-ADDRESS : uses
> ```

thanks

**Comment by tim-hub** [comment]
@laurent22 @coderrsid @tessus 

thanks for the finding out

**Comment by tessus** [comment]
No worries, at one point we'll update Mermaid. No ETA though.

**Comment by tessus** [comment]
@laurent22 I actually tried to update mermaid, but none of the scripts updates the mermaid in `rnInjectedJs`. What am I missing?

**Comment by tessus** [comment]
Mermaid has to be updated. Upstream already fixed the issue I encountered after upgrading to 8.5.0.
So we should be good to go.

I'll submit a PR tomorrow. I'm currently not home.

**Comment by tessus** [comment]
I've updated Mermaid to 8.5.2, but the squence diagram and the gantt chart are too small. I suspect that we have previously added additional css to Joplin to fix upstream issues, but I don't know.

@laurent22 do you remember adding fixes for Mermaid upstream bugs that had something to do with the size of the mermaid images?

<img width="312" alt="image" src="https://user-images.githubusercontent.com/223439/83979787-d05a3400-a8de-11ea-8aa8-861f364566a4.png">

**Comment by tessus** [comment]
@laurent22 do you remember adding fixes for Mermaid upstream bugs that had something to do with the size of the mermaid images?

**Comment by tim-hub** [comment]
> Hey there, it looks like there has been no activity on this issue recently. Has the issue been fixed, or does it still require the community's attention? This issue may be closed if no further activity occurs. You may comment on the issue and I will leave it open. Thank you for your contributions.

i think it still not fixed

**Comment by tessus** [comment]
The problem is that new versions of Mermaid require a change from fit-content to 100% for the width. But this also results in another issue described here: https://discourse.joplinapp.org/t/upgrading-mermaid/10140/4?u=tessus
However, since Laurent changed it to 100%, we can also upgrade Mermaid....

**Comment by laurent22** [comment]
Isn't it the same issue as this one? #3097 I think the fix would be to make the graph scrollable rather than 100% width, so that we can handle graphs of any size.

**Comment by tessus** [comment]
Well, the `fit-content` did that, but then some of the graphs are too small. I explained this in the topic referenced.
So I think at one point you added the `fit-content` but now after mermaid 8.5.0 some images are too small. Going back fixes that size problem, but reassets the scroll issue....

**Comment by MHoeller** [comment]
The issue is fixed, with version 1.2.x since this was a release I considered this as tested. 
In case it is needed: from my perspective this issue can be closed

**Comment by tessus** [comment]
Yep, Caleb added this with the bump to Mermaid 8.x.y
